### PR TITLE
Support Iceberg ingestion from REST based catalogs

### PR DESCRIPTION
**Pull request by a2l007**
### Description

- Adds support to the `iceberg` input source to read from Iceberg [REST](https://iceberg.apache.org/concepts/catalog/#decoupling-using-the-rest-catalog) catalogs.
- Upgrades iceberg core dependency to `1.6.1`

#### Release note

 Adds support to the `iceberg` input source to read from Iceberg REST Catalogs.
<hr>

<hr>

This PR has:

- [x] been self-reviewed.
- [x] added documentation for new or modified features or behaviors.
- [x] a release note entry in the PR description.
- [x] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [x] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [x] been tested in a test Druid cluster.

### CODE REVIEW

File: docs/ingestion/input-sources.md [hunk 1]
```diff
@@ -1063,7 +1063,7 @@ The following is a sample spec for a S3 warehouse source:
 
 ### Catalog Object
 
-The catalog object supports `local` and `hive` catalog types.
+The catalog object supports `local`,`hive` and `rest` catalog types.
```
Comment (abhishekrb19) [inline docs/ingestion/input-sources.md:1066]: Nit: maybe reorder the catalog sections by relevance/usage — rest, hive, and local?

File: extensions-contrib/druid-iceberg-extensions/src/test/java/org/apache/druid/iceberg/input/RestCatalogTest.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            httpExchange.sendResponseHeaders(200, outputBytes.length);
+            OutputStream os = httpExchange.getResponseBody();
+            httpExchange.getResponseHeaders().set(HttpHeaders.CONTENT_TYPE, "application/octet-stream");
+            httpExchange.getResponseHeaders().set(HttpHeaders.CONTENT_LENGTH, String.valueOf(outputBytes.length));
+            httpExchange.getResponseHeaders().set(HttpHeaders.CONTENT_RANGE, "bytes 0");
+            os.write(outputBytes);
+            os.close();
+          }
+      );
+      server.start();
```
Comment (abhishekrb19) [inline extensions-contrib/druid-iceberg-extensions/src/test/java/org/apache/druid/iceberg/input/RestCatalogTest.java:66]: This should probably be in test `setup()`?

**Review by abhishekrb19 — COMMENTED** [review]
Looks good overall, thanks! A few suggestions.

File: docs/ingestion/input-sources.md [hunk 3]
```diff
@@ -1084,9 +1084,18 @@ The following table lists the properties of a `hive` catalog:
...
+|Property|Description|Default|Required|
+|--------|-----------|-------|---------|
+|type|Set this value to `rest`.|None|yes|
+|catalogUri|The URI associated with the catalog's HTTP endpoint.|None|yes|
+|catalogProperties|Map of any additional properties that needs to be attached to the catalog.|None|no|
+
 ### Iceberg filter object
 
 This input source provides the following filters: `and`, `equals`, `interval`, and `or`. You can use these filters to filter out data files from a snapshot, reducing the number of files Druid has to ingest.
+If the filter column is not an Iceberg partition column, it is highly recommended to define an additional filter defined in the [`transformSpec`](./ingestion-spec.md#transformspec). This is because for non-partition columns, Iceberg filters may return rows that do not match the expression.
```
[the remark this answers is not in the document]
Comment (a2l007) [reply docs/ingestion/input-sources.md]: Thanks for the comments.
Curious, have you encountered any cases where the `transformSpec` doesn't fully guarantee additional filtering on top of the delta lake filters?
Transform spec filters have worked out well for us in this case and so I'm keen to understand if there are any gotchas.

[the remark this answers is not in the document]
Comment (abhishekrb19) [reply docs/ingestion/input-sources.md]: Ah, good to know. I was just imagining a scenario where the `transformSpec` filters in Druid don't map 1:1 to a native filter that Delta Lake supports. I did a quick [search](https://druid.apache.org/docs/latest/querying/filters), and it seems they can be mapped to the ones we already support at least, so I think we should be good.

**Review by abhishekrb19 — APPROVED** [review]

**Pull request merged**